File: modules/agent/src/models/OrganizationTechnology.py
"""OrganizationTechnology model for tech stack junction table."""

from sqlalchemy import Column, Text, DateTime, BigInteger, Numeric, ForeignKey, func, Integer
from sqlalchemy.orm import relationship

from models import Base


class OrganizationTechnology(Base):
    """OrganizationTechnology SQLAlchemy model (junction table)."""

    __tablename__ = "Organization_Technology"

    organization_id = Column(BigInteger, ForeignKey("Organization.id", ondelete="CASCADE"), primary_key=True)
    technology_id = Column(BigInteger, ForeignKey("Technology.id", ondelete="CASCADE"), primary_key=True)
    detected_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    source = Column(Text, nullable=True)       # 'builtwith', 'wappalyzer', 'agent', 'manual'
    confidence = Column(Numeric(3, 2), nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
    created_by = Column(BigInteger, ForeignKey("User.id"), nullable=False)
    updated_by = Column(BigInteger, ForeignKey("User.id"), nullable=False)

    # Relationships
    organization = relationship("Organization", back_populates="technologies")
    technology = relationship("Technology")

File: modules/agent/internal/models/OrganizationTechnology.go
package models

import (
	"time"

	"github.com/shopspring/decimal"
)

type OrganizationTechnology struct {
	OrganizationID int64            `gorm:"primaryKey" json:"organization_id"`
	TechnologyID   int64            `gorm:"primaryKey" json:"technology_id"`
	DetectedAt     time.Time        `gorm:"autoCreateTime" json:"detected_at"`
	Source         *string          `json:"source"` // builtwith, wappalyzer, agent, manual
	Confidence     *decimal.Decimal `gorm:"type:numeric(3,2)" json:"confidence"`
	CreatedAt      time.Time        `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt      time.Time        `gorm:"autoUpdateTime" json:"updated_at"`
}

func (OrganizationTechnology) TableName() string {
	return "Organization_Technology"
}
